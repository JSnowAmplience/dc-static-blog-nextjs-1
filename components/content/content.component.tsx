import Image from '../images/image.component';
import Video from '../videos/video.component';
import CodeBlock from '../code-block/code-block.component';
import { AmplienceContent } from '../../common/interfaces/content.type';
import ReactMarkdown from 'react-markdown';
import InlineCode from '../inline-code/inline-code.component';

const MARKDOWN_RENDERERS = {
  code: CodeBlock,
  inlineCode: InlineCode
};

const Content = ({ content }: { content: AmplienceContent[] }) => {
  return (
    <>
      <section>
        {content.map((c: AmplienceContent) => {
          if ('image' in c) {
            return (
              <div>
                <Image altText={c.altText} src={c.src} />
              </div>
            );
          } else if ('video' in c) {
            return (
              <div className="blog-post-video">
                <Video video={c.video} src={c.src} />
              </div>
            );
          } else if ('text' in c) {
            return (
              <div>
                <ReactMarkdown source={c.text} renderers={MARKDOWN_RENDERERS} />
              </div>
            );
          }
        })}
      </section>
      <style jsx>{`
        section {
          display: flex;
          justify-content: flex-start;
          flex-direction: column;
          align-items: flex-start;
        }

        section div {
          width: 100%;
        }

        .blog-post-video :global(video) {
          width: 100%;
        }

        .blog-post-code {
          width: 100%;
        }

        :global(blockquote) {
          padding: 10px;
          border-left: 10px solid #ccc;
          font-size: 1rem;
          font-style: italic;
          font-weight: 300;
          color: #666;
        }

        :global(table) {
          border-collapse: collapse;
        }

        :global(th),
        :global(td) {
          border: 1px solid #666;
          padding: 5px;
        }

        :global(.inline-code) {
          padding: 3px;
          border: 1px solid pink;
          border-radius: 5px;
          background-color: #eee;
        }
      `}</style>
    </>
  );
};

export default Content;
