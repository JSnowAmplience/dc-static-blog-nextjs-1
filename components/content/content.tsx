import Image from '../images/image.component';
import Video from '../videos/video.component';
import { AmplienceContent } from '../../common/interfaces/content.type';
import ReactMarkdown from 'react-markdown';
import markdown from '../markdown-renderers/markdown';
import theme from '../../common/styles/default/theme';

const MARKDOWN_RENDERERS = { ...markdown };

const Content = ({ content }: { content: AmplienceContent[] }) => {
  return (
    <>
      <section>
        {content.map((c: AmplienceContent, index: number) => {
          if ('image' in c) {
            return (
              <div key={c.image.id}>
                <Image
                  altText={c.altText}
                  src={c.src}
                  dynamicImagingOptions={[{ w: 675, scaleFit: 'poi' }, { w: 374, scaleFit: 'poi' }]}
                />
              </div>
            );
          } else if ('video' in c) {
            return (
              <div key={c.video.id} className="blog-post-video">
                <Video video={c.video} srcSet={c.srcSet} />
              </div>
            );
          } else if ('text' in c) {
            return (
              <div key={`text${index}`}>
                <ReactMarkdown source={c.text} renderers={MARKDOWN_RENDERERS} />
              </div>
            );
          }
        })}
      </section>
      <style jsx>{`
        section {
          color: ${theme.colors.doveGray};
          display: flex;
          justify-content: flex-start;
          flex-direction: column;
          align-items: flex-start;
        }

        section > div {
          margin-top: 15px;
          width: 100%;
        }

        section > div:last-child {
          margin-bottom: 0;
        }

        section :global(table) {
          border-collapse: collapse;
        }

        section :global(th),
        section :global(td) {
          border: 1px solid ${theme.colors.silver};
          padding: 12px;
        }

        section :global(tr:nth-child(odd) td) {
          background-color: ${theme.colors.whiteLilac};
        }

        section :global(p:first-child) {
          margin-top: 0;
        }

        section :global(p:last-child) {
          margin-bottom: 0;
        }

        @media (max-width: ${theme.layout.narrowPageWidth}) {
          section > div:first-child {
            margin-top: 15px;
          }

          section :global(pre) {
            font-size: ${theme.fonts.size.small};
          }
        }
      `}</style>
    </>
  );
};

export default Content;
