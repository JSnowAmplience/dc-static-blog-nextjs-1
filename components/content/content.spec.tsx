import ShallowRenderer from 'react-test-renderer/shallow';
import Content from './content';
import { AmplienceContent } from '../../common/interfaces/content.type';

describe('Content', (): void => {
  let renderer;
  beforeEach(() => {
    renderer = ShallowRenderer.createRenderer();
  });
  it('should render different types of content - text, image and video', (): void => {
    const content = [
      {
        image: {
          defaultHost: 'i1-qa.adis.ws',
          endpoint: 'bloblogltd',
          name: 'casual-wear',
          id: 'e1b511d2-1a33-47e7-8dc7-f460534cb0c7'
        },
        altText: 'Casual crotch shot'
      },
      {
        text:
          '# This is my first Bloblog\n\nThis is some sample text.\n\n## Look-a-header\n\nThis is some more sample text'
      },
      {
        video: {
          defaultHost: 'i1-qa.adis.ws',
          endpoint: 'bloblogltd',
          name: 'SampleVideo_1280x720_5mb',
          id: '721044de-d125-4a1a-8ddc-2201b9463f2d'
        },
        srcSet: ['http://i1-qa.adis.ws/v/bloblogltd/SampleVideo_1280x720_5mb/mp4_240p']
      }
    ];

    renderer.render(<Content content={content} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
