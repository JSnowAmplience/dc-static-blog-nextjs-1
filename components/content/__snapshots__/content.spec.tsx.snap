// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render different types of content - text, image and video 1`] = `
<React.Fragment>
  <section
    className="jsx-538569914"
  >
    <div
      className="jsx-538569914"
    >
      <Image
        altText="Casual crotch shot"
        dynamicImagingOptions={
          Array [
            Object {
              "w": 675,
            },
            Object {
              "w": 374,
            },
          ]
        }
        src="//i1-qa.adis.ws/i/bloblogltd/casual-wear"
      />
    </div>
    <div
      className="jsx-538569914"
    >
      <ReactMarkdown
        astPlugins={Array []}
        escapeHtml={true}
        parserOptions={Object {}}
        plugins={Array []}
        rawSourcePos={false}
        renderers={
          Object {
            "blockquote": [Function],
            "code": [Function],
            "heading": [Function],
            "inlineCode": [Function],
            "link": [Function],
            "paragraph": [Function],
          }
        }
        skipHtml={false}
        source="# This is my first Bloblog

This is some sample text.

## Look-a-header

This is some more sample text"
        sourcePos={false}
        transformLinkUri={[Function]}
      />
    </div>
    <div
      className="jsx-538569914 blog-post-video"
    >
      <Video
        srcSet={
          Array [
            "http://i1-qa.adis.ws/v/bloblogltd/SampleVideo_1280x720_5mb/mp4_240p",
          ]
        }
        video={
          Object {
            "defaultHost": "i1-qa.adis.ws",
            "endpoint": "bloblogltd",
            "id": "721044de-d125-4a1a-8ddc-2201b9463f2d",
            "mediaType": "v",
            "name": "SampleVideo_1280x720_5mb",
          }
        }
      />
    </div>
  </section>
  <JSXStyle
    dynamic={
      Array [
        "#666666",
        "#cccccc",
        "#cccccc",
        "#f7f7fc",
        "736px",
      ]
    }
    id="1039621377"
  >
    section.__jsx-style-dynamic-selector{color:#666666;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:start;-webkit-justify-content:flex-start;-ms-flex-pack:start;justify-content:flex-start;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:flex-start;-webkit-box-align:flex-start;-ms-flex-align:flex-start;align-items:flex-start;padding-bottom:50px;border-bottom:1px solid #cccccc;}section.__jsx-style-dynamic-selector&gt;div.__jsx-style-dynamic-selector{margin-top:75px;width:100%;}section.__jsx-style-dynamic-selector table{border-collapse:collapse;}section.__jsx-style-dynamic-selector th,section.__jsx-style-dynamic-selector td{border:1px solid #cccccc;padding:12px;}section.__jsx-style-dynamic-selector tr:nth-child(odd) td{background-color:#f7f7fc;}@media (max-width:736px){section.__jsx-style-dynamic-selector&gt;div.__jsx-style-dynamic-selector{margin-top:90px;}}
  </JSXStyle>
</React.Fragment>
`;
